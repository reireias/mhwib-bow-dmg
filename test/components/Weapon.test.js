import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import target from '@/components/Weapon'
import { createStore } from '@/test/helper'

Vue.use(Vuetify)
const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

describe('instance', () => {
  let store
  beforeEach(() => {
    const getters = {
      calcuratedWeapon: () => ({ attack: 240, affinity: 0, element: 300 }),
    }
    store = createStore(getters)
  })

  it('should be vue instance', () => {
    const vuetify = new Vuetify()
    const wrapper = shallowMount(target, {
      localVue,
      store,
      vuetify,
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
