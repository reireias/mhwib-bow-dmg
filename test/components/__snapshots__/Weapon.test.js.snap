// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instance should be vue instance 1`] = `
"<v-container-stub tag=\\"div\\">
  <v-row-stub tag=\\"div\\">
    <v-col-stub tag=\\"div\\">
      <div class=\\"headline\\">武器</div>
    </v-col-stub>
  </v-row-stub>
  <v-row-stub tag=\\"div\\">
    <v-col-stub tag=\\"div\\">
      <bow-stub></bow-stub>
    </v-col-stub>
  </v-row-stub>
  <customs-stub></customs-stub>
  <v-row-stub tag=\\"div\\">
    <v-col-stub tag=\\"div\\">
      <parts-stub></parts-stub>
    </v-col-stub>
  </v-row-stub>
  <v-row-stub tag=\\"div\\">
    <v-col-stub tag=\\"div\\">
      <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" readonly=\\"true\\" rules=\\"\\" successmessages=\\"\\" value=\\"240\\" backgroundcolor=\\"\\" label=\\"攻撃力\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" reverse=\\"true\\" type=\\"text\\"></v-text-field-stub>
    </v-col-stub>
    <v-col-stub tag=\\"div\\">
      <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" readonly=\\"true\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" label=\\"武器倍率\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" reverse=\\"true\\" type=\\"text\\"></v-text-field-stub>
    </v-col-stub>
    <v-col-stub tag=\\"div\\">
      <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" readonly=\\"true\\" rules=\\"\\" successmessages=\\"\\" value=\\"0\\" backgroundcolor=\\"\\" label=\\"会心率\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" reverse=\\"true\\" type=\\"text\\"></v-text-field-stub>
    </v-col-stub>
    <v-col-stub tag=\\"div\\">
      <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" readonly=\\"true\\" rules=\\"\\" successmessages=\\"\\" value=\\"300\\" backgroundcolor=\\"\\" label=\\"属性値\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" reverse=\\"true\\" type=\\"text\\"></v-text-field-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>"
`;
